packages/flutter_tools/gradle/src/main/kotlin/DependencyVersionChecker.kt

```diff
@@ -157,43 +171,60 @@ object DependencyVersionChecker {
         }
     }
 
+    @VisibleForTesting internal fun getSDKVersion(project: Project): Int? {
+        val androidExtensionName = "android"
+        val androidExtension =
+            project.extensions.findByName(androidExtensionName) as? ApplicationExtension
```

Comment:
Please use the type safe approach instead
```suggestion
            project.extensions.findByType(ApplicationExtension::class.java)
```